src/test/opentrackIO_parser/python/opentrackIOlib.py

```diff
@@ -0,0 +1,304 @@
+            schema_units = self.sd["properties"]["timing"]["properties"]["sampleTimestamp"]["units"]
+            print("Schema says sample time units are: {}".format(schema_units))
+            print("Setting preferred sample time format to: {0}".format(format_str))
+        if (format_str in self.sample_time_formats):
+            self.sample_time_format = format_str
+
+    def Set_focus_distance_units(self,unit_str):
+        """Establish a user-preference for units of focus distance by storing a conversion factor.
+        Valid: m, cm, mm, in"""
+        schema_units = self.sd["properties"]["lens"]["properties"]["focusDistance"]["units"]
+        if self.verbose:
+            print("Schema says focus distance units are: {}".format(schema_units))
+            print("Setting preferred focus distance units to: {}".format(unit_str))
+        if schema_units == "millimeter":
+            if unit_str == "m":
+                self.focus_dist_mult = .001
+            elif unit_str == "cm":
+                self.focus_dist_mult = 0.1
+            elif unit_str == "mm":
+                self.focus_dist_mult = 1.0
+            elif unit_str == "in":
+                self.focus_dist_mult = 1/25.4
+
+    def Get_protocol(self):
+        """The protocol to which this sample conforms"""
+        #if (self.pd["protocol"]):
+            #return str(self.pd["protocol"])
+        if (self.pd["protocolVersion"]):
+            return str(self.pd["protocolVersion"])
+        else:
+            return None
+
+    def Get_slate(self):
+        """The current slate denoting scene,setup,take etc."""
+        if "device" in self.pd.keys():
+            if self.pd["device"]["slate"]:
+                return str(self.pd["device"]["slate"])
+            else:
+                return None
+        else:
+            return None
+
+    def Get_sensor_dimension_height(self):
+        """Return the height of the camera sensor.
+        If present in this sample, the 'static' block would have this info"""
+        if "static" in self.pd: 
+            if self.pd["static"]["camera"]["activeSensorResolution"]:
+                height = int(self.pd["static"]["camera"]["activeSensorResolution"]["height"])
+                return height
+        return None
+
+    def Get_sensor_dimension_width(self):
+        """Return the width of the camera sensor.
+        If present in this sample, the 'static' block would have this info"""
+        if "static" in self.pd: 
+            if self.pd["static"]["camera"]["activeSensorResolution"]:
+                width = int(self.pd["static"]["camera"]["activeSensorResolution"]["width"])
+                return width
+            else:
+                return None
+        return None
+
+    def Get_sensor_dimension_units(self):
+        """Returns the units found in the schema for active sensor dimensions"""
+        if "static" in self.pd: 
+            if self.sd["properties"]["camera"]["properties"]["activeSensorPhysicalDimensions"]:
```

Comment:
It would be good to use a key check helper here because this will throw KeyError if some part of the chain is missing...

Reply:
Done